src/notiDaemon/notiDaemon.vala

```diff
@@ -140,7 +146,7 @@ namespace SwayNotificationCenter {
                                         HashTable<string, Variant> hints,
                                         int expire_timeout) throws DBusError, IOError {
             uint32 id = replaces_id;
-            if (replaces_id == 0 || replaces_id > noti_id) id = ++noti_id;
+            if (replaces_id == 0 || replaces_id > action_index) id = ++action_index;
```

Comment:
This is supposed to be `noti_id` due to it actually being the id of the notification